src/core/lib/Magic.mjs

```diff
@@ -264,6 +291,35 @@ class Magic {
         return results;
     }
 
+    /**
+     *
+     */
+    checkRegexes(regexes) {
+        for (const elem of regexes) {
+            const regex = new RegExp(elem.match, elem.flags);
+            if  (regex.test(this.inputStr))
+                return true;
+        }
+        return false;
+    }
+    /**
+     *
+     */
+    checkOutputFromPrevious() {
+        let score = 0;
+        if ("regex" in this.prevOp.output) {
```

Comment:
This `in` operator usage looks way cleaner than the `Object.prototype.hasOwnProperty.call(object, key)` call. Please just do a mental check each time you use it to make sure you won't be hitting on something up the object's prototype chain. See [inherited properties here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in) for more info. (I can't see any bad uses here)